src/transport/msg_types/common.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from typing import Any, Callable, List
+
+from blockchain.typings import Web3
+from cryptography.verify_signature import recover_vs, verify_message_with_signature
+from eth_account.account import VRS
+from metrics.metrics import UNEXPECTED_EXCEPTIONS
+from transport.msg_types.deposit import DepositMessage
+from transport.msg_types.pause import PauseMessage
+from transport.msg_types.unvet import UnvetMessage
+from utils.bytes import from_hex_string_to_bytes
+
+logger = logging.getLogger(__name__)
+
+
+def get_messages_sign_filter(web3: Web3) -> Callable:
+    """Returns filter that checks message validity"""
+
+    def check_messages(msg: DepositMessage | PauseMessage | UnvetMessage) -> bool:
+        prefix = _message_prefix(web3, msg)
+        v, r, s = _vrs(msg)
+
+        data, abi = _verification_data(prefix, msg)
+
+        verified = verify_message_with_signature(
+            data=data,
+            abi=abi,
+            address=msg['guardianAddress'],
+            vrs=(v, r, s),
+        )
+
+        if not verified:
+            label_name = _select_label(msg)
+            logger.error({'msg': 'Message verification failed.', 'value': msg})
+            UNEXPECTED_EXCEPTIONS.labels(label_name).inc()
+
+        return verified
+
+    return check_messages
+
+
+def _vrs(msg: DepositMessage | PauseMessage | UnvetMessage) -> tuple[VRS, VRS, VRS]:
+    vs = msg['signature']['_vs']
+    if '_vs' in msg['signature']:
```

Comment:
Since '_vs' is required, let's remove this condition.
```suggestion
    r = msg['signature']['r']
    v, s = recover_vs(vs)
    return v, r, s
```

Reply:
Done